File: 1962. Remove Stones to Minimize the Total/Solution.java
import java.util.Arrays;
import java.util.PriorityQueue;

class Solution {
    public int minStoneSum(int[] piles, int k) {
        PriorityQueue<Integer> pq = this.getQueueFromArray(piles);
        int val;
        for (int i = 0; i < k; i++) {
            val = pq.poll();
            val -= Math.floor(val / 2);
            pq.add(val);
        }
        int sum = 0;
        for (Integer i : pq) {
            sum -= i;
        }
        return sum;
    }

    public PriorityQueue<Integer> getQueueFromArray(int[] array) {
        PriorityQueue<Integer> queue = new PriorityQueue<>();
        Arrays.stream(array).forEach((element) -> {
            queue.add(-element);
        });
        return queue;
    }

}

File: 1962. Remove Stones to Minimize the Total/Solution.py
import math
from queue import PriorityQueue
from typing import List


class Solution:
    def minStoneSum(self, piles: List[int], k: int) -> int:
        pq = PriorityQueue()
        for pile in piles:
            pq.put(-pile)
        for i in range(k):
            val = pq.get()
            val = val+math.floor(-val/2)
            pq.put(val)
        sum = 0
        while not pq.empty():
            val = pq.get()
            print(val)
            sum = sum-val
        return sum
